### ISSUE TITLE
Technical concerns with vanity-subdomains

### ISSUE THREAD

**Issue by bytaesu** [supabase/supabase#30237]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

There is something unusual with setting up a vanity subdomain using the Supabase CLI. I believe that any subdomain is being marked as available. Previously, after activating a subdomain on my instance and running check-availability, it showed as unavailable. However, now it still shows as available even after being set up, which seems strange.  To confirm this behavior, I tested multiple subdomains, and I am confident this is not a coincidence but a bug.

## Expected behavior

Potentially conflicting vanity subdomains, or those that could be considered Supabase resources (e.g., api.supabase.co), are currently being marked as available for activation.

## Screenshots
<img width="576" alt="a" src="https://github.com/user-attachments/assets/26cb6e19-b7c8-4f7a-9b54-853c877cc7be">
<img width="562" alt="api" src="https://github.com/user-attachments/assets/261927f0-072d-4b10-b522-5e30214e2662">
<img width="573" alt="apple" src="https://github.com/user-attachments/assets/fb1dfcb9-ebc6-448b-92db-8ae26b024da9">
<img width="558" alt="x" src="https://github.com/user-attachments/assets/bee70562-b279-4aa8-8883-bdc524707db2">
<img width="575" alt="google" src="https://github.com/user-attachments/assets/4c18b929-7db3-4cbc-9d1b-4adea2c73751">

## System information

- OS: macOS

## Additional context

I have not yet explored how vanity subdomains are configured within Supabase. If this is indeed a bug and submitting a PR would help resolve the issue, I am willing to contribute one. (Though I hope it gets resolved before then.)

**Comment by cyluss** [comment]
So your point is the command must validate the input and disallow reserved subdomain names? If so please update the issue title and 'Expected behavior' section accordingly.
I guess the vanity subdomain feature is internal to Supabase's hosted platform, so it's not something external contributors can work on. https://github.com/supabase/cli/blob/4cf189afa6dc8b9738b1de31e4c06fcb6d5e1aaf/internal/vanity_subdomains/check/check.go#L25

**Comment by bytaesu** [comment]
What I mean is that I’m concerned about whether the vanity subdomain availability check is functioning properly. 

A few months ago, out of curiosity, I checked widely used subdomains, and at that time, the result indicated that they were unavailable. However, now, everything I tested shows as available (even when I check subdomains I have already set up as mine). 

I created an issue because I’m concerned about whether this is working correctly or not.
